plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -0,0 +1,520 @@
+	 */
+	private function log_production_warning() {
+		WP_CLI::log( __( 'This feature is not production ready yet. Make sure you are not running these commands in your production environment.', 'woocommerce' ) );
+	}
+
+	/**
+	 * Count how many orders have yet to be migrated into the custom orders table.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp wc cot count_unmigrated
+	 *
+	 * @param array $args Positional arguments passed to the command.
+	 *
+	 * @param array $assoc_args Associative arguments (options) passed to the command.
+	 *
+	 * @return int The number of orders to be migrated.*
+	 */
+	public function count_unmigrated( $args = array(), $assoc_args = array() ) : int {
+		if ( ! $this->is_enabled() ) {
+			return 0;
+		}
+
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		$order_count = $this->synchronizer->get_current_orders_pending_sync_count();
+
+		$assoc_args = wp_parse_args(
+			$assoc_args,
+			array(
+				'log' => true,
+			)
+		);
+		if ( isset( $assoc_args['log'] ) && $assoc_args['log'] ) {
+			WP_CLI::log(
+				sprintf(
+					/* Translators: %1$d is the number of orders to be migrated. */
+					_n( 'There is %1$d order to be migrated.', 'There are %1$d orders to be migrated.', $order_count, 'woocommerce' ),
+					$order_count
+				)
+			);
+		}
+
+		return (int) $order_count;
+	}
+
+	/**
+	 * Migrate order data to the custom orders table.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--batch-size=<batch-size>]
+	 * : The number of orders to process in each batch.
+	 * ---
+	 * default: 500
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp wc cot migrate --batch-size=500
+	 *
+	 * @param array $args Positional arguments passed to the command.
+	 * @param array $assoc_args Associative arguments (options) passed to the command.
+	 */
+	public function migrate( $args = array(), $assoc_args = array() ) {
+		$this->log_production_warning();
+		if ( ! $this->is_enabled() ) {
+			return;
+		}
+
+		if ( $this->synchronizer->custom_orders_table_is_authoritative() ) {
+			return WP_CLI::error( __( 'Migration is not yet supported when custom tables are authoritative. Switch to post tables as authoritative source if you are testing.', 'woocommerce' ) );
+		}
+
+		$order_count = $this->count_unmigrated();
+
+		// Abort if there are no orders to migrate.
+		if ( ! $order_count ) {
+			return WP_CLI::warning( __( 'There are no orders to migrate, aborting.', 'woocommerce' ) );
+		}
+
+		$assoc_args  = wp_parse_args(
+			$assoc_args,
+			array(
+				'batch-size' => 500,
+			)
+		);
+		$batch_size  = ( (int) $assoc_args['batch-size'] ) === 0 ? 500 : (int) $assoc_args['batch-size'];
+		$progress    = WP_CLI\Utils\make_progress_bar( 'Order Data Migration', $order_count / $batch_size );
+		$processed   = 0;
+		$batch_count = 1;
+		$total_time  = 0;
+
+		while ( $order_count > 0 ) {
+
+			WP_CLI::debug(
+				sprintf(
+				/* Translators: %1$d is the batch number, %2$d is the batch size. */
```

Comment:
Minor thing (indentation):

```suggestion
					/* Translators: %1$d is the batch number, %2$d is the batch size. */
```